server/types/record.go

```diff
@@ -51,5 +55,11 @@ var Record = &DoltgresType{
 	Acl:           nil,
 	Checks:        nil,
 	attTypMod:     -1,
-	CompareFunc:   toFuncID("btrecordcmp", toInternal("record"), toInternal("record")),
+}
```

Comment:
We should leave the `CompareFunc`, but instead make it `toFuncID("-")` which will indicate that it's not set.